src/App.tsx

```diff
@@ -8,7 +7,45 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+
+  const [query, setQuery] = useState('');
+  const [selectFilter, setSelectFilter] = useState('all');
+  const [activeTodoId, setActiveTodoId] = useState(0);
+
+  const activeTodo = todos.find(todo => todo.id === activeTodoId);
+
+  const getTodosList = useCallback(async (promise: Promise<Todo[]>) => {
+    try {
+      setTodos(await promise);
+    } catch {
+      setIsError(true);
+    }
+  }, []);
+
+  let VisibleTodos = todos;
+
+  if (selectFilter === 'active') {
+    VisibleTodos = VisibleTodos.filter(todo => !todo.completed);
+  }
+
+  if (selectFilter === 'completed') {
+    VisibleTodos = VisibleTodos.filter(todo => todo.completed);
+  }
```

Comment:
Maybe better use switch/case for that + create enum instead of comparing to strings?